### PR TITLE
Update CHANGELOG

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
+- New dptensor submodule containing numpy_usm_shared(dparray).
+- Examples of building Cython extensions with DPC++ compiler, that interoperate with dpCtl.
+- Callback mechanism so that numba_dppy.numpy_usm_shared can register with dpctl.dptensor.numpy_usm_shared.ndarray a callback function to look and see if the object is a Numba MemInfo with USM allocator.
+
+### Changed
+- Renamed DPPL to DPCTL.
+- Refactoring the backend functions to prep for changes to add L0 support.
+- Moved SyclProgram and SyclKernel out of dpctl into dpctl.program.
+
 ## [Unreleased]
```
Comment (PokhodenkoSA) [inline CHANGELOG.md:15]: Move it up above released version.

**Comment by vlad-perevezentsev** [comment]
Do I need to specify these changes?
https://github.com/IntelPython/dpctl/pull/185

**Comment by diptorupd** [comment]
> 
> 
> Do I need to specify these changes?
> #185

We can skip these changes as they only impact internal API of dpctl-capi. Otherwise, the changelog looks good. Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,18 @@ All notable changes to this project will be documented in this file.
...
 - setup.py builds C++ backend for develop and install commands.
 
+## [0.5.0] - 2020-12-17
+### Added
+- New dptensor submodule containing `numpy_usm_shared`(dparray).
+- Examples of building Cython extensions with DPC++ compiler that interoperate with dpCtl.
+- Callback mechanism so that `numba_dppy.numpy_usm_shared` can register with `dpctl.dptensor.numpy_usm_shared.ndarray` a callback function to look and see if the object is a Numba MemInfo with USM allocator.
+
+### Changed
+- Renamed DPPL to DPCTL.
```
Comment (diptorupd) [inline CHANGELOG.md:15]: We should add more details here. How about: "All usage of `DPPL` in C-API functions was changed to `DPCTL`, _e.g._, `DPPLQueueMgr_GetCurrentQueue` to `DPCTLQueueMgr_GetCurrentQueue`.

Comment (diptorupd) [reply CHANGELOG.md:15]: Were there any changes to the Python API functions? If so, please add a similar line with an example.

**Review by diptorupd — APPROVED** [review]

**Comment by diptorupd** [comment]
@PokhodenkoSA this is good to go.

**Pull request merged**